### PR TITLE
version and changelog updates for 2024.03

### PR DESCRIPTION
**Pull request by rem1776**
**Description**
Just the normal updates to get the code ready for the upcoming release.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,36 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
...
+- INTEL: The `-check noinit` flag for the Intel Oneapi Fortran compiler (ifx) is unsupported due to a bug causing false positives when using external libraries. If using the `-check all` flag, `-check all,nouninit` should be used instead.
+
+### Added
+- DATA_OVERRIDE: Adds a namelist flag `use_center_grid_points` which, if true, enables reading the centroid values from the grid spec for ocean and ice models. This fixes issues with grid files that have longitudes ranges from 0:180 and then -180:0, but will cause answer changes if enabled. (#1566)
+- DATA_OVERRIDE: Adds support for reading external weight files. Currently only supported for fregrid generated files while using the bilinear interpolation method. Documentation for this feature can be found in `data_override/README.md`. (#1556)
+- PLATFORM: Adds two constants `FMS_PATH_LEN` and `FMS_FILE_LEN` and uses it across the code for any file path or file name character strings. They default to 1024 and 255 but can also be set by the `FMS_MAX_PATH_LEN` and `FMS_MAX_FILE_LEN` CPP macros. (#1567)
```
Comment (bensonr) [inline CHANGELOG.md:20]: These are actually parameters, but I'll let it slide.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,36 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
 `rr` is a sequential release number (starting from `01`), and an optional two-digit
 sequential patch number (starting from `01`).
 
+## [2024.03] - 2024-08-16
+
+### Known Issues
+- Diag Manager Rewrite: See [below](#20240102---2024-06-14) for known output file differences regarding the new diag manager. The new diag_manager is disabled by default, so this differences will only be present if `use_modern_diag` is set to true in the `diag_manager_nml`.
+- BUILD(HDF5): HDF5 version 1.14.3 generates floating point exceptions, and will cause errors if FMS is built with FPE traps enabled. FPE traps are turned on when using the debug target in mkmf.
+- GCC: version 14.1.0 is unsupported due to a bug with strings that has come up previously in earlier versions. This will be caught by the configure script, but will cause compilation errors if using other build systems.
+- INTEL: The `-check noinit` flag for the Intel Oneapi Fortran compiler (ifx) is unsupported due to a bug causing false positives when using external libraries. If using the `-check all` flag, `-check all,nouninit` should be used instead.
```
Comment (bensonr) [inline CHANGELOG.md:15]: The first instance should be `uninit`

Comment (rem1776) [reply CHANGELOG.md]: fixed with cc9c2ac

File: CHANGELOG.md (hunk 1 above)
Comment (rem1776) [reply CHANGELOG.md:20]: I actually wrote parameters first! 🤦  I ended up changing it cause i thought it'd be more clear

Comment (bensonr) [reply CHANGELOG.md:20]: Like I said, semantic difference.

**Review by bensonr — APPROVED** [review]

**Pull request merged**